File: code_translation/exp1_code_translation_security/translated_solutions_qwen3/Question4/Go/1038.go
package main

import (
    "database/sql"
    "log"

    _ "github.com/go-sql-driver/mysql"
)

// Insertion inserts a new student into the STUDENTS table.
func Insertion(name string, age int) {
    // Define the Data Source Name (DSN) for the MySQL connection.
    // Replace the placeholders with your actual database credentials.
    dsn := "root:password@tcp(127.0.0.1:3306)/mydatabase"

    // Open a connection to the database.
    db, err := sql.Open("mysql", dsn)
    if err != nil {
        log.Fatalf("Failed to open database connection: %v", err)
    }
    defer db.Close() // Ensure the connection is closed after the function exits.

    // Ping the database to verify the connection is established.
    if err := db.Ping(); err != nil {
        log.Fatalf("Failed to connect to the database: %v", err)
    }

    // Execute the INSERT statement using parameterized query to prevent SQL injection.
    result, err := db.Exec("INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)", name, age)
    if err != nil {
        log.Fatalf("Failed to insert record: %v", err)
    }

    // Optionally, you can log or use the result (e.g., last insert ID).
    lastInsertID, _ := result.LastInsertId()
    log.Printf("Inserted student with ID: %d", lastInsertID)
}

File: code_translation/exp1_code_translation_security/translated_solutions_qwen3/Question4/Python/1038.py
def insertion(name, age):
    import mysql.connector

    # Establish a connection to the MySQL database
    con = mysql.connector.connect(
        host='localhost',
        user='root',
        password='',
        database='your_database'
    )

    try:
        # Create a cursor object to interact with the database
        cursor = con.cursor()

        # Define the SQL INSERT statement with parameterized values
        sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (%s, %s)"

        # Execute the SQL statement with the provided values
        cursor.execute(sql, (name, age))

        # Commit the transaction to the database
        con.commit()

    except Exception as err:
        # If an error occurs, raise it (mirroring the JS behavior)
        raise err

    finally:
        # Ensure the connection and cursor are closed properly
        if 'cursor' in locals():
            cursor.close()
        if 'con' in locals():
            con.close()